src/local/butler/reproduce_tool/android.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Android emulator installation and management."""
+from __future__ import print_function
+
+import os
+import time
+
+from local.butler.reproduce_tool import errors
+from local.butler.reproduce_tool import prompts
+from platforms.android import adb
+from platforms.android import device
+from system import environment
+from system import new_process
+
+ADB_DEVICES_SEPARATOR_STRING = 'List of devices attached'
+EMULATOR_PATH = os.path.join(
+    os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, os.pardir,
+    'local', 'bin', 'android-sdk', 'emulator', 'emulator')
+
+
+def start_emulator():
+  """Return a ProcessRunner configured to start the Android emulator."""
+  runner = new_process.ProcessRunner(
+      EMULATOR_PATH,
+      ['-avd', 'Android28', '-writable-system', '-partition-size', '2048'])
+  emulator_process = runner.run()
+
+  # If we run adb commands too soon after the emulator starts, we may see
+  # flake or errors. Delay a short while to account for this.
+  # TODO(mbarbella): This is slow and flaky, but wait-for-device isn't usable if
+  # another device is connected (as we don't know the serial yet). Find a better
+  # solution.
+  time.sleep(30)
```

Comment:
can we start the emulator with a predictable serial?

https://developer.android.com/studio/run/emulator-commandline says there's 

```
 -prop ro.serialno=123456
```

Reply:
++++1, predictable serial will be nice, user will be unhappy with 30 sec wait, too long.

Reply:
Great suggestion. Thanks!

Reply:
Unfortunately this doesn't seem to change how ADB recognizes the device. After setting that it still only recognized it as "emulator-5554", and running commands with -s 123456 failed.